pkg/controller/job/job_controller.go

```diff
@@ -1353,15 +1353,18 @@ func (jm *Controller) removeTrackingFinalizerFromPods(ctx context.Context, jobKe
 
 // enactJobFinished adds the Complete or Failed condition and records events.
 // Returns whether the Job was considered finished.
-func (jm *Controller) enactJobFinished(job *batch.Job, finishedCond *batch.JobCondition) bool {
-	if finishedCond == nil {
+func (jm *Controller) enactJobFinished(logger klog.Logger, jobCtx *syncJobCtx) bool {
+	if jobCtx.finishedCondition == nil {
 		return false
 	}
+	job := jobCtx.job
 	if uncounted := job.Status.UncountedTerminatedPods; uncounted != nil {
-		if len(uncounted.Succeeded) > 0 || len(uncounted.Failed) > 0 {
+		if count := len(uncounted.Succeeded) + len(uncounted.Failed); count > 0 {
+			logger.V(2).Info("Delaying marking the Job as finished, because there are still uncounted pod(s)", "job", klog.KObj(job), "condition", jobCtx.finishedCondition.Type, "count", count)
```

Comment:
I think this is too high verbosity. The uncounted pods are tracked in the job status, so the user can infer the progress. If we want the logging, it should be at least level 4.

Reply:
Right, they are in the Job status indeed, but maybe for some OnCallers it will not be clear to be aware of that, so some log message could point in the right direction. OTOH, I'm ok with `V(4)` since this log wasn't even present in the previous versions

Reply:
Updated to V(4)